README.md

```diff
@@ -1,19 +1,92 @@
+For larger edits or new articles, you'll want to set up a local environment for
+editing. Please see our [Local Development
+document](./docs/local-development.md) to set that up.
 
-Docs for CircleCI Server Administration are built in a slightly different way so there is a [separate README](README-server-docs.md) for this section.
+If you have a question or need help debugging, please head to [CircleCI
+Discuss](https://discuss.circleci.com/) where our support team will help you
+out.
+
+## Documentation Components
+
+This repository houses and manages several arms of documentation for CircleCI.
+This section will provide a brief overview of each "component" and how to get 
+started with making changes.
+
+### `/Jekyll` - Main Site
+
+This is the main [CircleCI documentation site](https://circleci.com/docs/2.0/).
+This is built with Jekyll and houses the majority of our documentation. Other
+branches of documentation (`src-api`, `src-crg`, etc) eventually get moved into
+this folder (in our build process) and integrated into the Jekyll Site. Follow
+the [local development guide](./docs/local-development.md) to get started with
+building the Jekyll site.
+
+### `/src-api` - API V2 Build Tooling
+
+This is the build tooling we use for automatically generating documentation for
+the CircleCI API v2. This uses [Slate](https://github.com/slatedocs/slate) and
+[Widdershins](https://github.com/Mermade/widdershins) to create documentation
+with a spec (that follows the Open API Spec) generated from the CircleCI code
+base.
+
+### `/src-config-builder` - Configuration Builder
+
+The [configuration builder](https://circleci.com/docs/config-builder/) is a
+small React application that providers users with a basic configuration builder
+that they can access through our documentation. When it is deployed, the
+compiled site is moved into `/jekyll`.
+
+### `/src-crg` - Configuration Reference
+
+This is documentation for the 2.1 configuration keys, built with Slate and
+compiled and moved into the Jekyll site.
+
+### `/src-js` - Javascript Files
+
+Some JavaScript files that make use of Webpack  and bundles a Javascript
+file that eventually gets loaded into Jekyll. Our JavaScript files are
+splintered between Jekyll's assets folder and here. We should be navigating
+toward one solution that will eventually aggregate all JS source code in one place.
+
+### `/src-shared` - Shared Assets with circleci.com
+
+This is a *git sub-module* for shared content with the main [CircleCI
+website](https://circleci.com/docs/). The `js` folder within is symlinked into
+Jekyll's assets folder for JavaScript. Eventually, the JS here could/should be
+integrated with a better JS bundle solution per above. Please see the [local
+development](./docs/local-development.md) guide for more information about
+pulling in the updates for the sub-module.
+
+### Server Documentation
```

Comment:
All these new sections here are great!

Reply:
Thank you! There will be future cleanup, but this is a start!